FinancialTracker/ViewControllers/BalanceViewController.swift

```diff
@@ -10,16 +10,66 @@ import UIKit
 class BalanceViewController: UIViewController {
     @IBOutlet var balanceTextField: UITextField!
     @IBOutlet var nextButton: UIButton!
+    @IBOutlet var welcomeLabel: UILabel!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         balanceTextField.keyboardType = .numberPad
+                
+        FirebaseHandler.shared.getCurrentUserData(completionHandler: { firebaseError, success, data in
+            guard success == true else {
+                switch firebaseError {
+                case .some(let error):
+                    fatalError(error.localizedDescription)
+                case .none:
+                    break
+                }
+
+                return
+            }
+            guard let data = data else {
+                return
```

Comment:
Do not leave empty returns, use `assert(false)` before the `return`in the least, add a message where needed.